types/utils.go

```diff
@@ -21,16 +22,13 @@ var (
 	errUnsupportedKeyType = errors.New("unsupported key type")
 )
 
-// GetRandomValidatorSet returns a validator set with a single validator
-func GetRandomValidatorSet() *cmtypes.ValidatorSet {
-	valSet, _ := GetRandomValidatorSetWithPrivKey()
-	return valSet
-}
-
-// GetRandomValidatorSetWithPrivKey returns a validator set with a single
-// validator
-func GetRandomValidatorSetWithPrivKey() (*cmtypes.ValidatorSet, ed25519.PrivKey) {
-	privKey := ed25519.GenPrivKey()
+// GetValidatorSet returns a validator set with a single validator
+// with the given key. If privKey is present, it is returned back.
+// Otherwise, a new private key is generated and returned.
+func GetValidatorSet(privKey ed25519.PrivKey) (*cmtypes.ValidatorSet, ed25519.PrivKey) {
+	if privKey == nil {
+		privKey = ed25519.GenPrivKey()
+	}
```

Comment:
```suggestion
func GetRandomValidatorSet() (*cmtypes.ValidatorSet, ed25519.PrivKey) {
	privKey = ed25519.GenPrivKey()
```

Since we are making some changes to this helper functions I think we should update the naming here. 
We should have the default `Get` method not require any inputs and just generate it. 

Then we should update `GetRandomValidatorSetWithPrivKey` to `GetRandomVadlidatorSetCustom` and have `ValidatorConfig` as a type that has a `privkey` and `VotingPower`.

Then `GetRandomValidatorSet` just calls `GetRandomValidatorSetCustom` with the random privKey and voting power of 1.